### PR TITLE
[MNG-7587] Add an IT for jsr330 components

### PR DESCRIPTION
**Pull request by gnodet**
JIRA issue: https://issues.apache.org/jira/browse/MNG-7587
IT for https://github.com/apache/maven/pull/1119

### CODE REVIEW

File: core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7587Jsr330.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+import org.codehaus.plexus.util.Os;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledOnJre;
+import org.junit.jupiter.api.condition.JRE;
+
+import java.io.File;
+
+/**
+ * This is a test set for <a href="https://issues.apache.org/jira/browse/MNG-7737">MNG-7737</a>.
+ * Simply verifies that various (expected) profiles are properly activated or not.
```
Comment (laeubi) [inline core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7587Jsr330.java:33]: Javadoc seems wrong

File: core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7587Jsr330.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+     * Verify components can be written using JSR330 on JDK 1.8.
+     *
+     * @throws Exception in case of failure
+     */
+    @Test
+    @EnabledOnJre({JRE.JAVA_8, JRE.JAVA_11, JRE.JAVA_17})
+    void testJdk8()
+        throws Exception
+    {
+        File testDir = ResourceExtractor.simpleExtractResources( getClass(), "/mng-7587-jsr330");
```
Comment (laeubi) [inline core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7587Jsr330.java:56]: as code is always the same (except one parameter) maybe use a method that accepts the java version?!?

File: core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7587Jsr330.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+import org.apache.maven.shared.verifier.util.ResourceExtractor;
+import org.codehaus.plexus.util.Os;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.condition.EnabledOnJre;
+import org.junit.jupiter.api.condition.JRE;
+
+import java.io.File;
+
+/**
+ * This is a test set for <a href="https://issues.apache.org/jira/browse/MNG-7737">MNG-7737</a>.
```
Comment (cstamas) [inline core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7587Jsr330.java:32]: update javadoc

File: core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7587Jsr330.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    {
+        File testDir = ResourceExtractor.simpleExtractResources( getClass(), "/mng-7587-jsr330");
+
+        Verifier verifier = newVerifier( testDir.getAbsolutePath() );
+        verifier.addCliArgument( "clean" );
+        verifier.addCliArgument( "verify" );
+        verifier.addCliArgument( "-Drunning-java-release-version=17" );
+        verifier.execute();
+        verifier.verifyErrorFreeLog();
+    }
```
Comment (laeubi) [inline core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7587Jsr330.java:105]: Maybe java 20 can/should also be checked?

File: core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7587Jsr330.java [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+ * Simply verifies that various (expected) profiles are properly activated or not.
+ *
+ */
+class MavenITmng7587Jsr330
+    extends AbstractMavenIntegrationTestCase
+{
+
+    public MavenITmng7587Jsr330()
+    {
+        // affected Maven versions: 3.9.0
```
Comment (cstamas) [inline core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7587Jsr330.java:42]: this is probably leftover?

**Review by laeubi — APPROVED** [review]
Looks good!

File: core-it-suite/src/test/resources/mng-7587-jsr330/pom.xml [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+          <artifactId>maven-resources-plugin</artifactId>
+          <version>3.3.0</version>
+        </plugin>
+        <plugin>
+          <artifactId>maven-compiler-plugin</artifactId>
+          <version>3.10.1</version>
+        </plugin>
+        <plugin>
+          <artifactId>maven-plugin-plugin</artifactId>
+          <version>3.6.0</version>
```
Comment (slachiewicz) [inline core-it-suite/src/test/resources/mng-7587-jsr330/pom.xml:91]: 3.9.0

File: core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7587Jsr330.java [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+        pluginVerifier.addCliArgument( "-Drunning-java-release-version=" + jdk );
+        pluginVerifier.execute();
+        pluginVerifier.verifyErrorFreeLog();
+
+        final Verifier consumerVerifier = newVerifier( new File( testDir, "consumer" ).getPath() );
+        consumerVerifier.setLogFileName( "log-" + jdk + ".txt" );
+        consumerVerifier.addCliArgument( "clean" );
+        consumerVerifier.addCliArgument( "verify" );
+        consumerVerifier.addCliArgument( "-V" );
+        consumerVerifier.addCliArgument( "-Drunning-java-release-version=" + jdk );
```
Comment (laeubi) [inline core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7587Jsr330.java:103]: This seems oboslete for the consumer.

File: core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7587Jsr330.java [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+        pluginVerifier.setLogFileName( "log-" + jdk + ".txt" );
+        pluginVerifier.addCliArgument( "clean" );
+        pluginVerifier.addCliArgument( "install" );
+        pluginVerifier.addCliArgument( "-V" );
+        pluginVerifier.addCliArgument( "-Drunning-java-release-version=" + jdk );
+        pluginVerifier.execute();
+        pluginVerifier.verifyErrorFreeLog();
+
+        final Verifier consumerVerifier = newVerifier( new File( testDir, "consumer" ).getPath() );
+        consumerVerifier.setLogFileName( "log-" + jdk + ".txt" );
```
Comment (laeubi) [inline core-it-suite/src/test/java/org/apache/maven/it/MavenITmng7587Jsr330.java:99]: should this probabbly use a different logfile name?

**Pull request merged**